### ISSUE TITLE
sp_BlitzFirst does not create BlitzCache table

### ISSUE THREAD

**Issue by HakuAnnaruto** [BrentOzarULTD/SQL-Server-First-Responder-Kit#1268]
My Version of sp_BlitzFirst and sp_BlitzCache:

SET @Version = '5.9.5';
SET @VersionDate = '20171115';

Using the parameters below, run as a job step, the Job Agent (account) has full permissions. The "BlitzCache" table did not get created. I ran sp_BlitzCache standalone and created the table. Then re-ran the Job and all was well.

NOTE: the first time a ran the job I did NOT have the @OutputTableNameBlitzCache param. I added it and then the job blew up, so I ran sp_BlitzCache standalone.

use DBA_DB
go
Exec dbo.sp_blitzFirst @seconds=60
,@outputDatabasename = 'DBA_DB'
,@outputschemaname = 'dbo'
,@outputtablename='BlitzFirstResults'
,@outputTableNameFileSTats='BlitzFirstResults_FileStats'
,@outputTableNamePerfmonStats='BlitzFirstResults_PerfmonStatus'
,@outputTableNameWaitStats ='BlitzFirstResults_WaitStats'
,@OutputTableNameBlitzCache = 'BlitzCache'
,@OutputTableRetentionDays = 30;

SQL Server 2008 R2, W2K8 Server

Very first attempt. I have others to try and we'll see if things are different of if I'm doing something wrong.

**Comment by BlitzErik** [comment]
Did you get an error message when you ran BlitzCache?

**Comment by HakuAnnaruto** [comment]
By itself - no. As part of job/step -
setting up configuration variables [SQLSTATE 01000] (Message 50000)  Now starting diagnostic analysis [SQLSTATE 01000] (Message 50000)  Capturing first pass of wait stats, perfmon counters, file stats [SQLSTATE 01000] (Message 50000)  Beginning investigatory queries [SQLSTATE 01000] (Message 50000)  Finished running investigatory queries [SQLSTATE 01000] (Message 50000)  Waiting to match @Seconds parameter [SQLSTATE 01000] (Message 50000)  Capturing second pass of wait stats, perfmon counters, file stats [SQLSTATE 01000] (Message 50000)  Analyzing changes between first and second passes of DMVs [SQLSTATE 01000] (Message 50000)  Analysis finished, outputting results [SQLSTATE 01000] (Message 50000)  Calling sp_BlitzCache [SQLSTATE 01000] (Message 50000)  Invalid object name 'DBA_DB.dbo.BlitzCache'. [SQLSTATE 42S02] (Error 208).  The step failed.

**Comment by BlitzErik** [comment]
Okay, what happens when you run BlitzCache trying to log it to your table?

```
EXEC sp_BlitzCache      @OutputDatabaseName = 'DBA_DB',
                        @OutputSchemaName = 'dbo',
                        @OutputTableName = 'BlitzCache'
```

**Comment by HakuAnnaruto** [comment]
Checking @MinutesBack validity.
Creating temp tables for results and warnings.
Checking database validity
Checking sort order
Cleaning up old warnings for your SPID
Cleaning up old plans for your SPID
Creating temp tables for internal processing
Checking plan cache age
Setting up variables
Determining SQL Server version.
Creating dynamic SQL based on SQL Server version.
Ignoring system databases by default
Applying chosen sort order
Adding additional info columns for newer versions of SQL
Substituting NULLs for memory grant columns in older versions of SQL
Collecting execution plan information.
Attempting to aggregate stored proc info from separate statements
Warning: Null value is eliminated by an aggregate or other SET operation.
Computing CPU, duration, read, and write metrics
Update CPU percentage for stored procedures
Gather percentage information from grouped results
Begin XML nodes processing
Gathering high level plan information
Warning: Null value is eliminated by an aggregate or other SET operation.
Performing compile timeout checks
Performing compile memory limit exceeded checks
Performing unparameterized query checks
Performing index DML checks
Performing table DML checks
Gathering row estimates
Gathering statement costs
Updating statement costs
Gathering stored procedure costs
Checking for plan warnings
Checking for implicit conversion
Performing busy loops checks
Performing TVF join check
Checking for operator warnings
Checking for table variables
Checking for functions
Checking for expensive key lookups
Checking for expensive remote queries
Checking for expensive sorts
Checking for icky cursors
Checking for bad scans and plan forcing
Checking for computed columns that reference scalar UDFs
Checking for filters that reference scalar UDFs
Checking modification queries that hit lots of indexes
Checking for Spatial index use
Checking for wonky Index Spools
Performing query level checks
Gathering additional plan level information
Warning: Null value is eliminated by an aggregate or other SET operation.
Attempting to get stored procedure name for individual statements
Is Paul White Electric?
Getting information about implicit conversions and stored proc parameters
Setting configuration values
Checking for query level SQL Server issues.
Checking for forced parameterization and cursors.
Populating Warnings column
Populating Warnings column for stored procedures
Checking for plans with >128 levels of nesting
Checking for plans with no warnings
Writing results to table.

**Comment by BlitzErik** [comment]
Alright, so:

Did it write to a table?
Did the table get created?
Does the BlitzFirst process run correctly now?

**Comment by HakuAnnaruto** [comment]
As I mentioned in the beginning, the "BlitzCache" table did not get created until I ran the sp_BlitzCache standalone. I got the impression that running sp_BlitzFirst with ,@outputtablenameblitzcache = 'BlitzCache', the table would have been created and populated. I worked around that assumption by running sp_BlitzCache standalone. It's all-good, now. The job is running and table is populating. I only mentioned it here b/c, again, I got the impression that, outputtablenameblitzcache = 'BlitzCache' would have taken care of that. No worries.
